src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1442,7 +1444,10 @@ export default class DefaultDeviceController
         };
         track.addEventListener('ended', newDevice.endedCallback, { once: true });
       }
-      newDevice.groupId = this.getMediaTrackSettings(newDevice.stream)?.groupId || '';
+      newDevice.groupId = this.getGroupIdFromDeviceId(
+        kind,
+        this.getMediaTrackSettings(newDevice.stream)?.deviceId || ''
+      );
```

Comment:
So we are choosing the groupId from the device cache? Not sure if this would help though because the infinite loop is caused because the groupId that getUserMedia gives us is not within our device cache anyways. And should the source of truth for the groupId be the device cache or the stream we get from getUserMedia?

Reply:
Why does it not help? It should match the one in the cache in the next check and terminate the loop.

Reply:
@ltrung yeah, its hard to say because we're not sure why gUM returns this groupId that isn't listed by listAudioInputsDevices (in the case of the loop). If the groupId provided by gUM is not found in the device cache, then I suspect that the deviceId we get from gUM might not have a groupId within the device cache anyways. In that case, getGroupIdFromDeviceId might return ''. Correct me if I'm wrong.

Reply:
We only see this behavior with default so the deviceId should be there. If it is not then groupId = '' and the match check will terminate so no infinite loop.